python/cuml/solvers/qn.pyx

```diff
@@ -418,9 +418,42 @@ class QN(Base):
 
         return cudf.Series(preds)
 
+    def score(self, X, y):
+        from cuml.metrics import accuracy_score
+        return accuracy_score(y, self.predict(X))
+
     def __getattr__(self, attr):
-        if self.fit_intercept:
-            if attr == 'coef_':
-                return self.coef_[0:-1]
-            if attr == 'intercept_':
+        if attr == 'intercept_':
+            if self.fit_intercept:
                 return self.coef_[-1]
+            else:
+                return rmm.to_device(np.zeros(1))
```

Comment:
```suggestion
                return rmm.to_device(zeros(1))
```

Minor, but it's better to use the device `zeros` function https://github.com/rapidsai/cuml/blob/47606a516b52df369d14f8792b92919e1e2b59be/python/cuml/utils/numba_utils.py#L128